### PR TITLE
improve warning when nonunique colors used with manual_color_key

### LINKED ISSUE #1170: manual_color_key with non-unique colours throws unhelpful error
Using manual_color_key with non-unique colours throws the following unhelpful error.

```
BoundsError: attempt to access 1-element Array{Tuple{Measures.Length{:mm,Float64},Measures.Length{:mm,Float64}},1} at index [1:2]
macro expansion at dynamic.jl:67 [inlined]
(::Atom.##99#104)(::Dict{String,Any}) at eval.jl:74
handlemsg(::Dict{String,Any}, ::Dict{String,Any}, ::Vararg{Dict{String,Any},N} where N) at comm.jl:163
(::Atom.##11#13)() at event.jl:73
```

It took me a while to establish that the cause of this error was effectively due to the following generated code.

```julia
Guide.manual_color_key(legend_title, ["h", "Smooth"], ["black", "black"]),
```

In this case, I think it would be useful if Gadfly contained a sanity check.

### PR DESCRIPTION
**Pull request by tlnagy**
fixes #1170

### CODE REVIEW

**Comment by tlnagy** [comment]
Looks like tests are segfaulting on Travis. Not sure why. Tests are also segfaulting on https://github.com/GiovineItalia/Gadfly.jl/blob/f922f4d137a84cc9ff38e3cbd3b01847ca5b0a7e/test/testscripts/issue1125.jl#L23 locally for me on both this branch and master.

**Comment by bjarthur** [comment]
the travis errors are similar to https://github.com/GiovineItalia/Gadfly.jl/issues/1166.  i wonder whether it would happen with julia 0.6.2.

i can confirm the segfault with `issue1125.jl` locally.

File: src/guide.jl [hunk 1]
```diff
@@ -492,11 +492,18 @@ end
...
-    labels::Vector{AbstractString}
-    colors::Vector{C}
+    labels::OrderedDict{C, AbstractString}
+end
+function ManualColorKey(title, labels, colors::Vector{C}) where {C<:Color}
+    labeldict = OrderedDict{C, AbstractString}()
+    for (c, l) in zip(colors, labels)
+        if c in keys(labeldict)
+            error("Colors should not appear more than once")
+        end
```
Comment (bjarthur) [inline src/guide.jl]: `haskey(labeldict,c) || error()` would be more succint.  otherwise, lgtm once CI passes.

**Comment by bjarthur** [comment]
https://github.com/GiovineItalia/Gadfly.jl/issues/1123 might be related

**Comment by bjarthur** [comment]
worth noting that `issue1125.jl` works for me when run all by itself using `include("test/testscripts/issue1125.jl")`.   the failure only occurs for `Pkg.test("Gadfly")`.  so there must be something about the sequence of events leading up to it.

**Comment by bjarthur** [comment]
https://github.com/JuliaLang/julia/issues/27607 seems to have the same error, and [the fix](https://github.com/JuliaLang/julia/pull/27568) is scheduled to be [backported](https://github.com/JuliaLang/julia/pull/27621) to 0.6.4.

**Comment by tlnagy** [comment]
It looks like tests on v0.6 are passing on master again so I've updated this PR and hopefully CI passes this time.

**Comment by tlnagy** [comment]
Looks like they're passing now!

**Pull request merged**